optuna/pruners/hyperband.py

```diff
@@ -237,3 +241,51 @@ def __getattribute__(self, attr_name):  # type: ignore
                     return object.__getattribute__(self, attr_name)
 
         return _BracketStudy(study, bracket_index)
+
+
+class _HyperbandSampler(BaseSampler):
```

Comment:
How about `_BracketSampler`? I think it better describes its logic/better aligned with `_BracketStudy`.

Reply:
I think the name of `_BracketSampler` is not good because the `_HyperbandSampler` depends on the underlying `HyperbandPruner` but `_BracketStudy` depends on each bracket of `HyperbandPruner`. Therefore, I think `BracketStudy` and `_HyperbandSampler` are appropriate as they are now.